gadget/gadget.go

```diff
@@ -288,23 +288,13 @@ func systemOrSnapID(s string) bool {
 	return true
 }
 
-// ReadInfo reads the gadget specific metadata from gadget.yaml in the snap. If
-// constraints is nil, ReadInfo will just check for self-consistency, otherwise
-// rules for the classic or system seed cases are enforced.
-func ReadInfo(gadgetSnapRootDir string, constraints *ModelConstraints) (*Info, error) {
+// InfoFromMeta reads the provided gadget metadata. If constraints is nil, only the
+// self-consistency checks are performed, otherwise rules for the classic or
+// system seed cases are enforced.
+func InfoFromMeta(gadgetYaml []byte, constraints *ModelConstraints) (*Info, error) {
```

Comment:
(bike-sheed9 Maybe: `InfoFromGadgetYaml()` to match snap.infoFromSnapYaml? But no strong opinion either way.